managed/services/management/dbaas/dbaas_initializer.go

```diff
@@ -0,0 +1,251 @@
+}
+
+const defaultClusterName = "default-pmm-cluster"
+
+var errClusterExists = errors.New("cluster already exists")
+
+// NewInitializer returns initialized Initializer structure
+func NewInitializer(db *reform.DB, client dbaasClient, grafanaClient grafanaClient, versionService versionService) *Initializer {
+	l := logrus.WithField("component", "dbaas_initializer")
+	return &Initializer{
+		db:             db,
+		l:              l,
+		dbaasClient:    client,
+		grafanaClient:  grafanaClient,
+		versionService: versionService,
+	}
+}
+
+// Update updates current dbaas settings
+func (in *Initializer) Update(ctx context.Context) error {
+	settings, err := models.GetSettings(in.db)
+	if err != nil {
+		in.l.Errorf("Failed to get settings: %+v.", err)
+		return err
+	}
+	if settings.DBaaS.Enabled {
+		return in.Enable(ctx)
+	}
+	return in.Disable(ctx)
+}
+
+// Enable enables dbaas feature and connects to dbaas-controller
+func (in *Initializer) Enable(ctx context.Context) error {
+	in.m.Lock()
+	defer in.m.Unlock()
+	if in.enabled {
+		return nil
+	}
+	timeoutCtx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	err := in.dbaasClient.Connect(timeoutCtx)
+	cancel()
+	if err != nil {
+		return err
+	}
+	ctx, in.cancel = context.WithCancel(ctx)
+
+	in.enabled = true
+	kubeConfig, err := in.dbaasClient.GetKubeConfig(ctx, &dbaascontrollerv1beta1.GetKubeconfigRequest{})
+	if err == nil {
+		// If err is not equal to nil, dont' register cluster and fail silently
+		err := in.db.InTransaction(func(t *reform.TX) error {
+			cluster, err := models.FindKubernetesClusterByName(t.Querier, defaultClusterName)
+			if err != nil {
+				in.l.Errorf("failed finding cluster: %v", err)
+				return nil
+			}
+			if cluster != nil {
+				return errClusterExists
+			}
+			return nil
+		})
+		if err != nil {
+			if errors.Is(err, errClusterExists) {
+				return nil
+			}
+			return err
+		}
+
+		req := &dbaasv1beta1.RegisterKubernetesClusterRequest{
+			KubernetesClusterName: defaultClusterName,
+			KubeAuth: &dbaasv1beta1.KubeAuth{
+				Kubeconfig: kubeConfig.Kubeconfig,
+			},
+		}
+		_, err = in.RegisterCluster(ctx, req)
+		if err != nil {
+			return err
+		}
+		in.l.Info("Cluster is successfully initialized")
+	} else {
+		in.l.Errorf("failed getting kubeconfig inside cluster: %v", err)
+	}
+
+	return nil
+}
+
+// Disable disconnects from dbaas-controller and disabled dbaas feature
+func (in *Initializer) Disable(ctx context.Context) error {
+	in.m.Lock()
+	defer in.m.Unlock()
+	if !in.enabled { // Don't disable if already disabled
+		return nil
+	}
+	if in.cancel != nil {
+		in.cancel()
+	}
+	err := in.dbaasClient.Disconnect()
+	if err != nil {
+		return err
+	}
+	in.enabled = false
+	return nil
+}
+
+// RegisterCluster registers k8s cluster and installs all required operators
+func (in *Initializer) RegisterCluster(ctx context.Context, req *dbaasv1beta1.RegisterKubernetesClusterRequest) (*dbaasv1beta1.RegisterKubernetesClusterResponse, error) {
```

Comment:
TBH I don't like the idea that we moved this method here. It doesn't feel like it belongs here.

Reply:
What do you suggest?

Reply:
1. The best option would be to have layered architecture, so we wouldn't call a gRPC server. But it makes the code inconsistent with what we have in the project now.
2. Copy and paste?
3. Calling kubernetes_server from the initializer.